app/models/application.rb

```diff
@@ -1,4 +1,8 @@
 class Application < ApplicationRecord
   has_many :application_pets
   has_many :pets, through: :application_pets
+
+  validates :name, :street, :city, :state, :zip, :description, presence: true
+  validates :zip, length: { is: 5 }
+  validates :state, format: { with: /[A-Z]{2}/, message: "Please use 2 capital letters for the state abbreviation" }
```

Comment:
```suggestion
  validates :state, format: { with: /[A-Z]{2}/, message: "must be an abbreviation of two capital letters" }
```

Reply:
The grammar of `State Please use 2 capital letters for the state abbreviation` comes across as unnatural. I changed the error message up a little bit to make it flow better (also put this in the test validation).

I think ultimately, we could refactor state to be a dropdown of all 50 states (still only working with US pet adoption).